lib/Sema/CSApply.cpp

```diff
@@ -8148,6 +8149,95 @@ bool ConstraintSystem::applySolutionFix(
                                       fix.first.getArgumentLabels(*this),
                                       isa<SubscriptExpr>(call->getFn()));
   }
+
+  case FixKind::AddConformace: {
+    auto getMissingConformance = [&](ConstraintLocator *locator) {
+      auto *anchor = locator->getAnchor();
+      auto &requirement = locator->getPath().back();
+      auto result = MissingConformances.find({anchor, requirement.getValue()});
+      assert(result != MissingConformances.end());
+      return result->getSecond();
+    };
+
+    auto conformance = getMissingConformance(locator);
+
+    auto *anchor = locator->getAnchor();
+    auto owner = solution.simplifyType(getType(anchor));
+
+    auto type = conformance.first;
+    auto protocolType = conformance.second->getDeclaredType();
+
+    //  Find `ApplyExpr` based on a function expression attached to it.
+    auto findApplyExpr = [](Expr *parent, Expr *fnExpr) -> ApplyExpr * {
+      ApplyExpr *applyExpr = nullptr;
+      parent->forEachChildExpr([&applyExpr, &fnExpr](Expr *subExpr) -> Expr * {
+        auto *AE = dyn_cast<ApplyExpr>(subExpr);
+        if (!AE || AE->getFn() != fnExpr)
+          return subExpr;
+
+        applyExpr = AE;
+        return nullptr;
+      });
+      return applyExpr;
+    };
+
+    auto getArgumentAt = [](ApplyExpr *AE, unsigned index) -> Expr * {
+      assert(AE);
+
+      auto *arg = AE->getArg();
+      if (auto *TE = dyn_cast<TupleExpr>(arg))
+        return TE->getElement(index);
+
+      assert(index == 0);
+      if (auto *PE = dyn_cast<ParenExpr>(arg))
+        return PE->getSubExpr();
+
+      return arg;
+    };
+
+    auto *applyExpr = findApplyExpr(expr, anchor);
+
+    Optional<unsigned> atParameterPos;
+    // Sometimes fix is recorded by type-checking sub-expression
+    // during normal diagnostics, in such case call expression
+    // is unavailable.
+    if (applyExpr) {
+      // If this is a static, initializer or operator call,
+      // let's not try to diagnose it here, but refer to expression
+      // diagnostics.
+      if (isa<BinaryExpr>(applyExpr) || isa<TypeExpr>(anchor))
```

Comment:
I would have expected that these diagnostics would be better for operators (at least), but okay.

Reply:
There are some problems with them currently, where current way of listing overload choices is better IMO, otherwise it would just say something like `Int does not conform to RangeReplaceableCollection` :(